Source/GMCAbilitySystem/Private/Components/GMCAbilityComponent.cpp

```diff
@@ -449,30 +475,66 @@ void UGMC_AbilitySystemComponent::RPCConfirmAbilityActivation_Implementation(int
 	if (ActiveAbilities.Contains(AbilityID))
 	{
 		ActiveAbilities[AbilityID]->ServerConfirm();
-		UE_LOG(LogGMCAbilitySystem, VeryVerbose, TEXT("[RPC] Server Confirmed Ability Activation: %d"), AbilityID);
+		UE_LOG(LogGMCAbilitySystem, VeryVerbose, TEXT("[RPC] Server Confirmed Long-Running Ability Activation: %d"), AbilityID);
 	}
 }
 
-TSubclassOf<UGMCAbility> UGMC_AbilitySystemComponent::GetGrantedAbilityByTag(FGameplayTag AbilityTag)
+TArray<TSubclassOf<UGMCAbility>> UGMC_AbilitySystemComponent::GetGrantedAbilitiesByTag(FGameplayTag AbilityTag)
 {
 	if (!GrantedAbilityTags.HasTag(AbilityTag))
 	{
 		UE_LOG(LogGMCAbilitySystem, Warning, TEXT("Ability Tag Not Granted: %s"), *AbilityTag.ToString());
-		return nullptr;
+		return {};
 	}
 
 	if (!AbilityMap.Contains(AbilityTag))
 	{
 		UE_LOG(LogGMCAbilitySystem, Warning, TEXT("Ability Tag Not Found: %s | Check The Component's AbilityMap"), *AbilityTag.ToString());
-		return nullptr;
+		return {};
 	}
 
-	return AbilityMap[AbilityTag];
+	return AbilityMap[AbilityTag].Abilities;
 }
 
 void UGMC_AbilitySystemComponent::InitializeAbilityMap(){
-	if(!StartingAbilityMap) return;
-	AbilityMap.Append(StartingAbilityMap.Get()->GetAbilityMap());
+	for (UGMCAbilityMapData* StartingAbilityMap : AbilityMaps)
+	{
+		for (const FAbilityMapData Data : StartingAbilityMap->GetAbilityMapData())
+		{
+			AddAbilityMapData(Data);
+		}
+	}
+}
+
+void UGMC_AbilitySystemComponent::AddAbilityMapData(const FAbilityMapData& AbilityMapData)
+{
+	if (AbilityMap.Contains(AbilityMapData.InputTag))
+	{
+		AbilityMap[AbilityMapData.InputTag] = AbilityMapData;
```

Comment:
could be helpful to have some way to append to this rather than just replacing

Reply:
Hmm, it does support multiple abilities being activated from one tag... I guess we can append and make it the responsibility of the user that if they don't want that behavior, they need to clean up the old tag.